htlcswitch/link.go

```diff
@@ -1487,26 +1497,37 @@ func (l *channelLink) ShortChanID() lnwire.ShortChannelID {
 // within the chain.
 //
 // NOTE: Part of the ChannelLink interface.
-func (l *channelLink) UpdateShortChanID(sid lnwire.ShortChannelID) {
-	l.Lock()
-	defer l.Unlock()
+func (l *channelLink) UpdateShortChanID(sid lnwire.ShortChannelID) error {
+	chanID := l.ChanID()
+
+	// Refresh the channel state's short channel ID by loading it from disk.
+	// This ensures that the channel state accurately reflects the updated
+	// short channel ID.
+	err := l.channel.State().RefreshShortChanID()
```

Comment:
This will fetch the updated ID from the DB, but doesn't look like this is used? Shouldn't we ensure `l.channel.ShortChanID() == l.shortChanID` when this method returns?

Reply:
my general thinking is that we should actually not pass in a short chan id at all. instead we can just refresh and pass back the current short chan id to the switch. this eliminates the potential scenario where the passed short chan id and the refreshed one don't match. it's not really clear what the recovery procedure would be in that case anyway

Reply:
fwiw, i started to make this change but realized it made our mock link's more difficult to mock. currently we pass in the short chan id we want them to return next, with this we'd need to have an additional layer of abstraction backing the mock link to simulate the refresh

Reply:
Alright, ended up changing this so that we use the value loaded from Refresh and not pass it in explicitly. Thought it was going to sprawl more, but alas not :)